app/src/main/java/com/expidev/gcmapp/MeasurementDetailsActivity.java

```diff
@@ -613,4 +808,48 @@ protected void onPostExecute(MeasurementDetails measurementDetails)
             onLoadDetailsFromServer(measurementDetails);
         }
     }
+
+    private class SaveMeasurementsToServer extends AsyncTask<MeasurementDetails, Void, Boolean>
+    {
+        @Override
+        protected Boolean doInBackground(MeasurementDetails... params)
+        {
+            MeasurementDetails measurementDetails = params[0];
+            GmaApiClient apiClient = GmaApiClient.getInstance(MeasurementDetailsActivity.this);
+
+            try
+            {
+                List<MeasurementDetails> data = new ArrayList<>();
+                data.add(measurementDetails);
+
+                Log.i(TAG, "Saving new measurements to the database");
+                MeasurementsService.saveMeasurementDetailsToDatabase(MeasurementDetailsActivity.this, measurementDetails);
+
+                Log.i(TAG, "Sending new measurements to the server");
+                return apiClient.updateMeasurementDetails(data, preferences.getString(PREF_CURRENT_ASSIGNMENT, null));
```

Comment:
this should have access to the currentAssignment object in `MeasurementsDetailsActivity`. Although that could lead to a race condition since that value is changed on the UI thread and doInBackground runs on a background Executor thread.

maybe pass in the current assignment either when creating the task: `new SaveMeasurementsToServer(currentAssignment).execute(data)`